mkosi/config.py

```diff
@@ -795,6 +794,57 @@ def cache_manifest(self) -> dict[str, Any]:
         return manifest
 
 
+def parse_ini(path: Path, sections: Sequence[str] = ()) -> Iterator[tuple[str, str, str]]:
+    section = setting = value = None
+
+    for l in textwrap.dedent(path.read_text()).splitlines():
+        comment = l.find("#")
+        if comment >= 0:
+            l = l[:comment]
+
+        if not l.strip():
+            continue
+
+        # If we have a section, setting and value, any line that's indented is considered part of the
+        # setting's value.
+        if section and setting and value is not None and l[0].isspace():
+            value = f"{value}\n{l.strip()}"
+            continue
+
+        # So the line is not indented, that means we either found a new section or a new setting. Either way,
+        # let's yield the previous setting and its value before parsing the new section/setting.
+        if section and setting and value is not None:
+            yield section, setting, value
+            setting = value = None
+
+        l = l.strip()
+
+        if l[0] == '[':
+            if l[-1] != ']':
+                die(f"{l} is not a valid section")
+
+            section = l[1:-1]
+            continue
+
+        if not section:
+            die(f"Setting {l} is located outside of section")
+
+        if sections and section not in sections:
```

Comment:
How about calling `sections` `only_sections` instead, right now this is a bit of a tongue twister to read.